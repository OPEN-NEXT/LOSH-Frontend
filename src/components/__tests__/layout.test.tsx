import site from "../../../gatsby-config";
// ES6
import Layout from "../layout";
import { useStaticQuery } from "gatsby";
import * as React from "react";
import ShallowRenderer from "react-test-renderer/shallow";

const renderer = ShallowRenderer.createRenderer();

jest.mock("antd", () => {
	const Layout = (): null => null;
	Layout.Content = (): null => null;

	return {
		Layout,
		Grid: {
			useBreakpoint: (): unknown => ({})
		}
	};
});

jest.mock("../header", () => {
	const Header = (): null => null;
	return Header;
});
jest.mock("../footer", () => {
	const Footer = (): null => null;
	return Footer;
});

beforeEach(() => {
	useStaticQuery.mockImplementationOnce(() => ({
		site
	}));
});

describe("Layout", () => {
	it("renders correctly", () => {
		const tree = renderer.render(<Layout></Layout>);
		expect(tree).toMatchSnapshot();
	});
});
