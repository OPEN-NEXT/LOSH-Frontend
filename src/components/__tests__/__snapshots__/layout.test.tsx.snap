// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly 1`] = `
<UNDEFINED
  style={
    Object {
      "margin": "auto",
      "maxWidth": 1440,
    }
  }
>
  <Header
    headerLinks={
      Array [
        Object {
          "title": "Explore Data",
          "to": "/",
        },
        Object {
          "title": "Contribute Data",
          "to": "/contribute-data",
        },
        Object {
          "title": "Ecosystem",
          "to": "/ecosystem",
        },
        Object {
          "title": "Get involved",
          "to": "/get-involved",
        },
      ]
    }
    siteTitle="Library of Open Source Hardware"
  />
  <Content />
  <Footer
    footerLinks={
      Array [
        Object {
          "title": "(c) OPEN!NEXT",
          "to": "/",
        },
        Object {
          "title": "About the Project",
          "to": "/about",
        },
        Object {
          "title": "Legal Disclaimer",
          "to": "/disclaimer",
        },
        Object {
          "title": "Submit an issue",
          "to": "/issue",
        },
      ]
    }
  />
</UNDEFINED>
`;
